### DISCUSSION TITLE
How to test GRPC client?

### DISCUSSION THREAD

**Discussion by softshipper** [quarkusio/quarkus#26204] [Q&A]
Hi all,

We are using GRPC client in our Quarkus based application, which does the call to an external GRPC server to get some data.
Now, we would like to test our Quarkus based application, which means we need somehow to mock the GRPC server. 

Our question is, how to mock a GRPC server in Quarkus for testing purposes? 

Kind reagars

[a comment by an automated account is not shown]

**Comment by michalszynkiewicz** [comment]
I think the simplest thing you can do is actually implement the grpc service in your test, i.e. add a grpc service to your test classes.
Did you try it?
